src/image/p5.Image.js

```diff
@@ -828,4 +831,87 @@ p5.Image.prototype.save = function(filename, extension) {
   p5.prototype.saveCanvas(this.canvas, filename, extension);
 };
 
+p5.Image.prototype.play = function() {
+  if (this.gifProperties === null || this.gifProperties.playing === true) {
+    return;
+  } else {
+    this.gifProperties.playing = true;
+  }
+};
+
+p5.Image.prototype.pause = function() {
+  if (this.gifProperties === null || this.gifProperties.playing === false) {
+    return;
+  } else {
+    this.gifProperties.playing = false;
+  }
+};
+
+p5.Image.prototype.delay = function(d) {
+  if (this.gifProperties === null) {
+    console.error('This is not a gif');
+    return;
+  } else {
+    this.gifProperties.delay = d;
+  }
+};
+
+p5.Image.prototype._createGIF = function(
+  arrayBuffer,
+  successCallback,
+  failureCallback,
+  finished
+) {
+  var gifReader = new omggif.GifReader(arrayBuffer);
+  var frames = [];
+  var numFrames = gifReader.numFrames();
+  var loadGIFFrameIntoImage = function(frameNum, gifReader) {
+    var framePixels = [];
+    try {
+      gifReader.decodeAndBlitFrameRGBA(frameNum, framePixels);
+    } catch (e) {
+      p5._friendlyFileLoadError(8, this.src);
+      if (typeof failureCallback === 'function') {
+        failureCallback(e);
+      } else {
+        console.error(e);
+      }
+    }
+    return new Uint8ClampedArray(framePixels);
+  };
+
+  for (var i = 0; i < numFrames; i++) {
+    var framePixels = loadGIFFrameIntoImage(i, gifReader);
+    var imageData = new ImageData(framePixels, this.width, this.height);
+    frames.push(imageData);
+  }
+
+  //Uses Netscape block encoding
+  //to repeat forever, this will be 0
+  //to repeat just once, this will be null
+  //to repeat N times (1<N), should contain integer for loop number
+  var loopCount = gifReader.loopCount();
+  if (loopCount === null) {
+    loopCount = 1;
+  }
+
+  this.gifProperties = {
+    isGif: true,
+    displayIndex: 0,
+    delay: gifReader.frameInfo(0).delay * 10,
```

Comment:
what is the `* 10` for?

Reply:
added comment explaining - basically gifs store delay in 1/100 of second so changing to millis here